src/script/lua_api/l_nodetimer.cpp

```diff
@@ -98,37 +86,21 @@ int NodeTimerRef::l_get_elapsed(lua_State *L)
 {
 	MAP_LOCK_REQUIRED;
 	NodeTimerRef *o = checkobject(L, 1);
-	ServerEnvironment *env = o->m_env;
-	if(env == NULL) return 0;
-
-	NodeTimer t = env->getMap().getNodeTimer(o->m_p);
+	NodeTimer t = o->m_map->getNodeTimer(o->m_p);
 	lua_pushnumber(L,t.elapsed);
 	return 1;
 }
 
-
-NodeTimerRef::NodeTimerRef(v3s16 p, ServerEnvironment *env):
-	m_p(p),
-	m_env(env)
-{
-}
-
 // Creates an NodeTimerRef and leaves it on top of stack
 // Not callable from Lua; all references are created on the C side.
-void NodeTimerRef::create(lua_State *L, v3s16 p, ServerEnvironment *env)
+void NodeTimerRef::create(lua_State *L, v3s16 p, ServerMap *map)
 {
-	NodeTimerRef *o = new NodeTimerRef(p, env);
+	NodeTimerRef *o = new NodeTimerRef(p, map);
 	*(void **)(lua_newuserdata(L, sizeof(void *))) = o;
 	luaL_getmetatable(L, className);
 	lua_setmetatable(L, -2);
 }
 
-void NodeTimerRef::set_null(lua_State *L)
```

Comment:
This was unused, right?

Reply:
if not the compilation failed no ? :D 
yes, and i don't see the idea to drop the m_env pointer from the nodetimerref, do you have a usecase for that, which will trigger the way that each thing you will do on nodetimer will return nil ?
As you may see there is not line removal on the lua api registration

Reply:
The usecase would be to invalidate the reference if the `ServerEnvironment`/`Map` disappears, but that obviously doesn't happen.
So this is okay.